docs/ARCHITECTURE.md

```
@@ -9,106 +9,314 @@
+  class L3 dataproducts
+  class L4 streamers
+  class L5 pipelines
+  class IF interfaces
+  class UD userdata
+```
+
+<details>
+<summary><b>Legend</b></summary>
+
+| Layer | Description |
+|-------|-------------|
```

markdownlint-cli2 (0.20.0)

[warning] Table column style
Table pipe is missing space to the right for style "compact"

(MD060, table-column-style)